src/misc/lv_array.h

```diff
@@ -38,46 +40,161 @@ typedef struct {
+ * @return the number of elements stored in the array
+ */
+static inline uint32_t lv_array_size(const lv_array_t * array)
+{
+    return array->size;
+}
 
-void lv_array_destroy(lv_array_t * array);
+/**
+ * Return the capacity of the array, i.e. how many elements can be stored.
+ * @param array pointer to an `lv_array_t` variable
+ * @return the capacity of the array
+ */
+static inline uint32_t lv_array_capacity(const lv_array_t * array)
+{
+    return array->capacity;
+}
 
-bool lv_array_append(lv_array_t * array, const uint8_t * element);
+/**
+ * Return if the array is empty
+ * @param array pointer to an `lv_array_t` variable
+ * @return true: array is empty; false: array is not empty
+ */
+static inline bool lv_array_is_empty(const lv_array_t * array)
+{
+    return array->size == 0;
+}
 
-uint8_t * lv_array_get(const lv_array_t * array, uint32_t index);
+/**
+ * Return if the array is full
+ * @param array pointer to an `lv_array_t` variable
+ * @return true: array is full; false: array is not full
+ */
+static inline bool lv_array_is_full(const lv_array_t * array)
+{
+    return array->size == array->capacity;
+}
 
-uint32_t lv_array_length(const lv_array_t * array);
+/**
+ * Copy an array to another.
+ * @note this will create a new array with the same capacity and size as the source array.
+ * @param target pointer to an `lv_array_t` variable to copy to
+ * @param source pointer to an `lv_array_t` variable to copy from
+ */
+void lv_array_copy(lv_array_t * target, const lv_array_t * source);
+
+/**
+ * Remove all elements in array.
+ * @param array pointer to an `lv_array_t` variable
+ */
+static inline void lv_array_clear(lv_array_t * array)
+{
+    array->size = 0;
+}
 
-uint32_t lv_array_capacity(const lv_array_t * array);
+/**
+ * Remove the element at the specified position in the array.
+ * @param array pointer to an `lv_array_t` variable
+ * @param index the index of the element to remove
+ * @return LV_RESULT_OK: success, otherwise: error
+ */
+lv_result_t lv_array_remove(lv_array_t * array, uint32_t index);
 
-bool lv_array_is_empty(const lv_array_t * array);
+/**
+ * Remove from the array either a single element or a range of elements ([start, end)).
+ * @note This effectively reduces the container size by the number of elements removed.
+ * @note When start equals to end, the function has no effect.
+ * @param array pointer to an `lv_array_t` variable
+ * @param start the index of the first element to be removed
+ * @param end the index of the first element that is not to be removed
+ * @return LV_RESULT_OK: success, otherwise: error
+ */
+lv_result_t lv_array_erase(lv_array_t * array, uint32_t start, uint32_t end);
 
-bool lv_array_is_full(const lv_array_t * array);
-/**********************
- *      MACROS
- **********************/
+/**
+ * Concatenate two arrays. Adds new elements to the end of the array.
+ * @note The destination array must be large enough to hold the result.
```

Comment:
Because the array is auto-resized when concatenating, we don't need this `@note` anymore. 
Or change to: _The destination array is automatically resized to hold the new items._